open-api/rest-catalog-open-api.yaml

```diff
@@ -2838,6 +2978,63 @@ components:
           additionalProperties:
             type: string
 
+    PreplanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+        - case-sensitive
+      properties:
+        select:
+          description: A list of the selected column names
+          type: array
+          items:
+            type: string
+        filter:
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Indicates whether column selection and filtering should be case sensitive
+          type: boolean
+        snapshot-id:
+          description: an int64 snapshot ID (if snapshot-range is not present); optional and defaults to the table's current snapshot
+          type: integer
+          format: int64
+        snapshot-range:
+          description: a JSON list containing exactly 2 int64 snapshot IDs (if snapshot-id is not present) representing the start (defaults to exclusive, see `start-exclusive`) and end (only inclusive) snapshots
+          type: array
+          items:
+            type: integer
+            format: int64
+        start-exclusive:
+          description: Indicates whether the user wants the start of `snapshot-range` to be inclusive or exclusive
+          type: boolean
+          default: true
+
+    PlanTableRequest:
+      type: object
+      required:
+        - select
+        - filter
+        - case-sensitive
+      properties:
+        select:
+          description: A list of the selected column names
```

Comment:
Can you add how to request nested fields? I had that in my original suggestion if you want to see how I think it would work.

Reply:
Sure can fix this. Is your original suggestion in the scan proposal google doc as a comment? If you can share it here as a comment would be helpful

Reply:
It was in my original comment.

Reply:
Can you share your original comment here? For some reason cant find this.

Reply:
Could you share your original comment in this thread, for some reason can not find it.

Reply:
@rdblue I believe this is the comment you are referring too, https://github.com/apache/iceberg/pull/9695#discussion_r1494914340?

Reply:
Look good. Thank you!